src/ai/backend/common/types.py

```diff
@@ -1142,6 +1142,13 @@ class EtcdRedisConfig(TypedDict, total=False):
     redis_helper_config: RedisHelperConfig
 
 
+def safe_print_redis_config(config: EtcdRedisConfig) -> str:
+    safe_config = config.copy()
+    if "password" in safe_config:
+        safe_config["password"] = "********"
+    return str(safe_config)
```

Comment:
The way you implemented it seems that safe_print_redis_config is optional.
How about processing the above logic in the `__str__` function of `EtcdRedisConfig` to make safe_print_redis_config the default behavior?
I think that password should be masked by default unless there is a special reason.

Reply:
> The way you implemented it seems that safe_print_redis_config is optional.
> How about processing the above logic in the `__str__` function of `EtcdRedisConfig` to make safe_print_redis_config the default behavior?
> I think that password should be masked by default unless there is a special reason.

As mentioned in the Teams thread, since `EtcdRedisConfig` is a `TypedDict`, we cannot add `__repr__` or `__str__` methods to this type.

So, as an alternative, I created this function.